test/test_gpuarray.py

```diff
@@ -862,6 +882,40 @@ def test_subset_minmax(self):
 
             assert min_a_gpu == min_a
 
+    def test_subset_sum(self):
+        """Test subset sum with annd without allocator"""
+
+        l_a = 2000
+        gran = 5
+        l_m = l_a - l_a // gran + 1
+
+        if has_double_support():
+            dtypes = [np.float64, np.float32, np.int32]
+        else:
+            dtypes = [np.float32, np.int32]
+
+        import pycuda.tools
+        for pool in [None, pycuda.tools.DeviceMemoryPool()]:
+            for dtype in dtypes:
+                a = np.random.uniform(0, 10, l_a).astype(dtype)
+                a_gpu = gpuarray.to_gpu(a)
+
+                meaningful_indices_gpu = gpuarray.zeros(l_m, dtype=np.int32)
+                meaningful_indices = meaningful_indices_gpu.get()
+                j = 0
+                for i in range(len(meaningful_indices)):
+                    meaningful_indices[i] = j
+                    j = j + 1
+                    if j % gran == 0:
+                        j = j + 1
+
+                meaningful_indices_gpu = gpuarray.to_gpu(meaningful_indices)
+
+                sum_a = a[meaningful_indices].sum()
+                allocator = None if pool is None else pool.allocate
+                sum_a_gpu = gpuarray.subset_sum(meaningful_indices_gpu, a_gpu, allocator=allocator).get()
+                assert np.allclose(sum_a_gpu, sum_a)
```

Comment:
Could you wrap the allocator (like so:
```py
alloc_uses =0
def allocator(size):
    nonlocal alloc_uses
    alloc_uses += 1
    return inner_alloc(size)
```
) to make sure it's actually getting used?

Reply:
OK ! Done in 88acc4e624